architecture-building-systems/CityEnergyAnalyst
# Add empty property to MultiChoiceParameter

This PR fixes #2966 by adding the `_.empty` property to `MultiChoiceParameter` which is inherited by `BuildingsParameter`. 
This will not only affect the `building` parameter, but all parameters in the config that uses `MultiChoiceParameter`. 

This solves the issue by keeping a state that indicates that the parameter is an empty value (selecting every value) when the parameter is initialized and reading from config. When getting the value of such parameter, it will just return every value if the empty state is `True`. The empty state will only be `False` when explicitly setting the parameter with a non empty value. 

So `BuildingsParameter` will remember the empty state instead of the buildings of the previous scenario when running workflow unless explicitly changed in later steps.

## Testing
Test workflow with 2 different scenarios with different number of buildings
- Run demand for each, with the scenario with **lesser** buildings before the other
    - the second scenario should still be able to run with all it's buildings
- Set the buildings parameter when running demand and check if as expected
